devel/ci/integration/tests/test_bodhi.py

```diff
@@ -16,16 +16,471 @@
+    # Fetch latest update with 'testing' status from DB
+    query_updates = (
+    # what about build.content, builds, bugs, testcases?
+        "SELECT "
+        "  updates.id as id, "
+        "  updates.alias as alias, "
+        "  updates.notes as notes, "
+        "  updates.status as status, "
+        "  updates.type as type, "
+        "  updates.severity as severity, "
+        "  updates.stable_karma as stable_karma, "
+        "  updates.unstable_karma as unstable_karma, "
+        "  updates.autokarma as autokarma, "
+        "  updates.request as request, "
+        "  updates.locked as locked, "
+        "  updates.suggest as suggest, "
+        "  users.name as username "
+        "FROM updates "
+        "JOIN users ON updates.user_id = users.id "
+        "WHERE status = 'testing' "
+        "ORDER BY date_submitted DESC LIMIT 1"
+    )
+    query_comments = (
+        "SELECT text FROM comments "
+        "WHERE update_id = %s"
+    )
+    db_ip = db_container.get_IPv4s()[0]
+    conn = psycopg2.connect("dbname=bodhi2 user=postgres host={}".format(db_ip))
+    with conn:
+        with conn.cursor() as curs:
+            curs.execute(query_updates)
+            update_info = {}
+            for value, description in zip(curs.fetchone(), curs.description):
+                update_info[description.name] = value
+            curs.execute(query_comments, (update_info["id"], ))
+            expected_comments = []
+            for row in curs.fetchall():
+                expected_comments.append(row[0])
+    conn.close()
+
+    # GET on latest testing update
+    with bodhi_container.http_client(port="8080") as c:
+        headers = {'Accept': 'text/html'}
+        http_response = c.get(f"/updates/{update_info['alias']}", headers=headers)
+
+    try:
+        assert http_response.ok
+        for key, value in update_info.items():
+            if key == "id":
+                pass
+            elif key in ["notes", "request"]:
+                if value != "" and value is not None:
+                    assert value.replace('\n', '') in http_response.text
+            elif key == "stable_karma":
+                assert f"stable threshold: {value}" in http_response.text
+            elif key == "unstable_karma":
+                assert f"unstable threshold: {value}" in http_response.text
+            elif key == "autokarma":
+                if value:
+                    assert "Enabled" in http_response.text
+                else:
+                    assert "Disabled" in http_response.text
+            elif key == "locked":
+                if value:
+                    assert "Locked" in http_response.text
+            elif key == "suggest":
+                if value == "reboot":
+                    assert "Reboot Required" in http_response.text
+                elif value == "logout":
+                    assert "Logout Required" in http_response.text
+            else:
+                assert value in http_response.text
```

Comment:
Instead of looping over the keys and values, why not assert the keys you expect directly?

Reply:
I'm not sure I understand. Do you think that it will be better to check for keys in http_response.text instead of values?

Reply:
I meant more to ask why we don't just expect the keys to be present and just make the above assertions, as opposed to the large if/else block. For example, why not do things like this instead:

```
if update_info['request']:
    assert update_info['request'] in http_response.text
assert f"stable threshold: {update_info['stable_karma']}" in http_response.text
<etc>
````

Reply:
At first, my intention was just to loop over values and check existance(of raw value) in response, but I find out that  most of `update's` fields are rendered different way than just raw value, so I started to adding more and more if/else statements. I will rewrite it.